sdk/python/packages/flet-core/src/flet_core/types.py

```diff
@@ -73,7 +74,13 @@ class NotchShape(Enum):
     CIRCULAR = "circular"
 
 
-ResponsiveNumber = Union[Dict[str, Union[int, float]], int, float]
+Number = Union[int, float]
+ResponsiveNumber = Union[Dict[str, Number], Number]
+OptionalNumber = Optional[Number]
+
+# str type alias
+ColorStr = Optional[str]
+IconStr = Optional[str]
 
```

Comment:
Love the idea of `ColorStr` and `IconStr`! Would it be possible/make sense to put that in *all* controls?

Reply:
Alternatively, we could convert `icons.py` and `colors.py` to Enums and use them for even better typing. Could be a separate issue/PR though.

Reply:
Yep, I think Enum is the way to go.

Reply:
> Alternatively, we could convert `icons.py` and `colors.py` to Enums and use them for even better typing. Could be a separate issue/PR though.

I can make in PR after merging